src/MovieList/MovieList.jsx

```diff
@@ -1,62 +1,73 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
 import { withRouter } from 'react-router';
+import { connect } from 'react-redux';
 
 import MovieAsset from './MovieAsset/MovieAsset';
 import styles from './MovieList.scss';
-import data from '../assets/data.json';
+import { getFilms } from '../actions';
 
 class MovieList extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      assets: [],
-    };
-  }
-
   componentDidMount = () => {
-    if (/^\/search$/.test(this.props.location.pathname)) {
-      const params = new URLSearchParams(this.props.location.search);
-      if (
-        params.get('name') !== null &&
-        params.get('name') !== '' &&
-        /^(title|director)$/.test(params.get('type'))
-      ) {
-        this.setState({
-          assets: data,
-        });
-      }
+    const { location } = this.props;
+    if (/^\/search\//.test(location.pathname)) {
+      this.checkQuery(location);
     }
   };
 
   componentWillReceiveProps = (nextProps) => {
     const { location } = this.props;
-    if (/^\/search$/.test(nextProps.location.pathname)) {
-      if (/^\/search$/.test(location.pathname) && nextProps.location.search !== location.search) {
-        const params = new URLSearchParams(nextProps.location.search);
-        if (
-          params.get('name') !== null &&
-          params.get('name') !== '' &&
-          /^(title|director)$/.test(params.get('type'))
-        ) {
-          this.setState({
-            assets: data,
-          });
-        } else {
-          this.state = {
-            assets: [],
-          };
-        }
+    const typeOfQuery = location.pathname.split('/')[2];
+    const newTypeOfQuery = nextProps.location.pathname.split('/')[2];
+    if (/^\/search\//.test(nextProps.location.pathname)) {
+      if (nextProps.location.search !== location.search || typeOfQuery !== newTypeOfQuery) {
+        this.checkQuery(nextProps.location);
       }
     }
   };
+
+  typeOfQuery;
+
+  checkQuery = (location) => {
+    const { dispatch } = this.props;
+    const params = new URLSearchParams(location.search);
+    if (params.get('query')) {
+      this.typeOfQuery = location.pathname.split('/')[2];
```

Comment:
Setting value to the object property will not guarantee you that component will be rerendered. So, use local state instead